src/search/components/timeframe.component.tsx

```diff
@@ -1,17 +1,142 @@
 import React from 'react';
-import { Box, Typography } from '@mui/material';
+import {
+  Box,
+  Divider,
+  Grid,
+  Typography,
+  Button,
+  TextField,
+} from '@mui/material';
 import { Schedule } from '@mui/icons-material';
 import { useClickOutside } from '../../hooks';
 
-const TimeframePopup = (): React.ReactElement => {
+export type TimeframeRange = {
+  value: number;
+  timescale: 'minutes' | 'hours' | 'days';
+};
+
+export interface TimeframeProps {
+  timeframe: TimeframeRange | null;
+  changeTimeframe: (value: TimeframeRange) => void;
+}
+
+const TimeframePopup = (props: TimeframeProps): React.ReactElement => {
+  const { changeTimeframe } = props;
+
+  const [workingTimeframe, setWorkingTimeframe] = React.useState<number>(0);
+
   return (
-    <div>
-      <Typography>Select your timeframe</Typography>
+    <div style={{ padding: 5 }}>
+      <Typography gutterBottom sx={{ fontWeight: 'bold' }}>
+        Select your timeframe
+      </Typography>
+      <Divider
+        sx={{
+          marginBottom: 2,
+          borderBottomWidth: 2,
+          backgroundColor: 'black',
+          width: '90%',
+        }}
+      />
+      <Grid container spacing={1} sx={{ paddingBottom: '15px' }}>
+        <Grid item xs={4}>
+          <Button
+            variant="outlined"
+            sx={{ height: '100%' }}
+            onClick={() => changeTimeframe({ value: 10, timescale: 'minutes' })}
+          >
+            Last 10 mins
+          </Button>
+        </Grid>
+        <Grid item xs={4}>
+          <Button
+            variant="outlined"
+            sx={{ height: '100%' }}
+            onClick={() => changeTimeframe({ value: 24, timescale: 'hours' })}
+          >
+            Last 24 hours
+          </Button>
+        </Grid>
+        <Grid item xs={4}>
+          <Button
+            variant="outlined"
+            sx={{ height: '100%' }}
+            onClick={() => changeTimeframe({ value: 7, timescale: 'days' })}
+          >
+            Last 7 days
+          </Button>
```

Comment:
Can we set `size="small"` on these buttons too? I don't think they need to be the bigger size, and it just adds a little consistency to all the button's padding etc in the popup

Reply:
I was debating that too, I'll go for it

Reply:
Addressed in https://github.com/ral-facilities/operationsgateway/pull/138/commits/a4a01470deab1f77015f18d40ad31b027b422d1d